components/TextFields/src/ContainedInputView/private/MDCTextControlVerticalPositioningReferenceBase.m

```diff
@@ -15,89 +15,187 @@
+      paddingBelowAssistiveLabelsRange * (1 - standardizedDensity);
+  _paddingBelowAssistiveLabels =
+      kMinPaddingBelowAssistiveLabels + paddingBelowAssistiveLabelsAddition;
+
+  CGFloat containerHeightWithPaddingsDeterminedByDensity = [self
+      calculateContainerHeightWithFoatingLabelHeight:floatingLabelHeight
+                                       textRowHeight:textRowHeight
+                                    numberOfTextRows:numberOfTextRows
+          paddingBetweenContainerTopAndFloatingLabel:_paddingBetweenContainerTopAndFloatingLabel
+           paddingBetweenFloatingLabelAndEditingText:_paddingBetweenFloatingLabelAndEditingText
+         paddingBetweenEditingTextAndContainerBottom:_paddingBetweenEditingTextAndContainerBottom];
+  if (preferredContainerHeight > 0) {
+    if (preferredContainerHeight > containerHeightWithPaddingsDeterminedByDensity) {
+      if (!isMultiline) {
+        CGFloat difference =
+            preferredContainerHeight - containerHeightWithPaddingsDeterminedByDensity;
+        CGFloat sumOfPaddingValues = _paddingBetweenContainerTopAndFloatingLabel +
+                                     _paddingBetweenFloatingLabelAndEditingText +
+                                     _paddingBetweenEditingTextAndContainerBottom;
+        _paddingBetweenContainerTopAndFloatingLabel =
+            _paddingBetweenContainerTopAndFloatingLabel +
+            ((_paddingBetweenContainerTopAndFloatingLabel / sumOfPaddingValues) * difference);
+        _paddingBetweenFloatingLabelAndEditingText =
+            _paddingBetweenFloatingLabelAndEditingText +
+            ((_paddingBetweenFloatingLabelAndEditingText / sumOfPaddingValues) * difference);
+        _paddingBetweenEditingTextAndContainerBottom =
+            _paddingBetweenEditingTextAndContainerBottom +
+            ((_paddingBetweenEditingTextAndContainerBottom / sumOfPaddingValues) * difference);
+      }
+    }
+  }
+
+  _containerHeight = containerHeightWithPaddingsDeterminedByDensity;
+  if (preferredContainerHeight > containerHeightWithPaddingsDeterminedByDensity) {
+    _containerHeight = preferredContainerHeight;
+  }
+
+  CGFloat halfOfNormalFontLineHeight = (CGFloat)0.5 * normalFontLineHeight;
+  if (isMultiline) {
+    CGFloat heightWithOneRow = [self
+        calculateContainerHeightWithFoatingLabelHeight:floatingLabelHeight
+                                         textRowHeight:textRowHeight
+                                      numberOfTextRows:1
+            paddingBetweenContainerTopAndFloatingLabel:_paddingBetweenContainerTopAndFloatingLabel
+             paddingBetweenFloatingLabelAndEditingText:_paddingBetweenFloatingLabelAndEditingText
+           paddingBetweenEditingTextAndContainerBottom:
+               _paddingBetweenEditingTextAndContainerBottom];
+    CGFloat halfOfHeightWithOneRow = (CGFloat)0.5 * heightWithOneRow;
+    _paddingBetweenContainerTopAndNormalLabel = halfOfHeightWithOneRow - halfOfNormalFontLineHeight;
```

Comment:
why is this not scaling with the others? using the normalized multiplier

Reply:
This chunk of code is determining where the label should go when it's not floating, i.e. when it's "normal." If it's single-line, it should be vertically centered within the container height. That's in the else clause. If it's multiline, it should go where it would go if it were single lined, even if the container height is taller. Does that make sense? I'm adding some comments and stuff that hopefully make it more clear.